### PR TITLE
javascript eval function

### PR DESCRIPTION
**Pull request by reyesjl**
implemented the JavaScript eval function

### CODE REVIEW

**Review by flashcode — CHANGES_REQUESTED** [review]

File: src/plugins/javascript/weechat-js.cpp [hunk 1]
```diff
@@ -63,6 +63,7 @@ struct t_plugin_script *js_current_script = NULL;
 struct t_plugin_script *js_registered_script = NULL;
 const char *js_current_script_filename = NULL;
 WeechatJsV8 *js_current_interpreter = NULL;
+char **js_buffer_output = NULL;
```
[the remark this answers is not in the document]
Comment (reyesjl) [reply src/plugins/javascript/weechat-js.cpp:66]: hmm, I see - will update tonight.

**Review by reyesjl — COMMENTED** [review]
Looks good to me.

**Comment by flashcode** [comment]
Hi,
Did you push your changes?
I don't see any changes in the commit.

**Comment by reyesjl** [comment]
Sebastien,

I did not. Which variable must I get rid of? - I was having trouble reading the information request and understanding which line or variable it was referring to do you think you can help me out in identifying this?

Thank you,
Jose

________________________________

**Comment by trygveaa** [comment]
> Sebastien, I did not. Which variable must I get rid of? - I was having trouble reading the information request and understanding which line or variable it was referring to do you think you can help me out in identifying this?

It's the `js_buffer_output` variable.

Did you compile this code? I'm getting these compilation errors when I try:

```
/home/trygve/dev/weechat/src/plugins/javascript/weechat-js.cpp: In function ‘void weechat_js_output_flush()’:
/home/trygve/dev/weechat/src/plugins/javascript/weechat-js.cpp:210:34: error: invalid conversion from ‘void*’ to ‘char*’ [-fpermissive]
  210 |                 command = malloc (length);
      |                           ~~~~~~~^~~~~~~~
      |                                  |
      |                                  void*
/home/trygve/dev/weechat/src/plugins/javascript/weechat-js.cpp: In function ‘int weechat_js_eval(t_gui_buffer*, int, int, const char*)’:
/home/trygve/dev/weechat/src/plugins/javascript/weechat-js.cpp:635:26: error: expected unqualified-id before ‘int’
  635 |     void *func_argv[1], *int result;
      |                          ^~~
/home/trygve/dev/weechat/src/plugins/javascript/weechat-js.cpp:641:41: error: ‘JS_EVAL_SCRIPT’ was not declared in this scope
  641 |                                         JS_EVAL_SCRIPT);
      |                                         ^~~~~~~~~~~~~~
/home/trygve/dev/weechat/src/plugins/javascript/weechat-js.cpp:655:5: error: ‘result’ was not declared in this scope
  655 |     result = weechat_js_exec (js_script_eval,
      |     ^~~~~~
make[2]: *** [src/plugins/javascript/CMakeFiles/javascript.dir/build.make:76: src/plugins/javascript/CMakeFiles/javascript.dir/weechat-js.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1698: src/plugins/javascript/CMakeFiles/javascript.dir/all] Error 2
```

**Comment by reyesjl** [comment]
hmm I apologize, I don't think I will be able to complete this.

**Pull request closed without merging**